rq/command.py

```diff
@@ -72,6 +72,7 @@ def handle_stop_job_command(worker, payload):
     job_id = payload.get('job_id')
     worker.log.debug('Received command to stop job %s', job_id)
     if job_id and worker.get_current_job_id() == job_id:
+        worker.get_current_job().set_status(JobStatus.STOPPED)
```

Comment:
I don’t like how we’re saving job status to Redis and is retrieved in `handle_job_failure()`. I think we can set a status to indicate that this job is being stopped in `worker` instance. We can probably set something like `worker.stopped_job_id = job_id` here and check this value during `handle_job_failure()` to know whether this job is commanded to stop.

Reply:
Is the objection mostly that the job status should always be set in the ```handle_job_*()``` methods and is out of place here?

Reply:
> Is the objection mostly that the job status should always be set in the handle_job_*() methods and is out of place here?

My main objection is that we're using Redis to pass status back and forth within the same process and job execution workflow. However, your observation is also valid (making status changes always set in `handle_job_*()` methods is also cleaner).

Reply:
@selwin I pushed a new commit, is this what you wanted?